### DISCUSSION TITLE
REST-like API

### DISCUSSION THREAD

**Discussion by chadwickalex** [gitmylo/audio-webui#54] [Ideas]
Are there any plans to implement some API endpoints that generate TTS and convert using RVC please? This would allow the project to be easily consumed by tools like oobabooga/text-generation-webui and SillyTavern/SillyTavern.
If you don't have any plans to do this yourself, would you accept a PR?

**Comment by gitmylo** [comment]
Currently only the gradio automatic API, and i don't have specific names for a few functions, i'll first give the common functions route names. But i could definitely also add a flask api. I'll probably make it as an extension once extensions are implemented, to show the flexibility of extensions!

**Reply by chadwickalex** [reply]
Looking at it right now and I'm not able to get the gradio API to work. Though it might just be me being dim. I think `fn_index=9`, which sets the tts model type, is not exposed and I'm getting errors attempting to imitate the `null` values for file path args which I can see in websocket response for `fn_index=10` which generates the tts output.
Appreciate it's not necessarily written to work this way, mostly trying to share info for anyone else looking

**Reply by gitmylo** [reply]
yeah, text to speech definitely isn't made for an API, since it takes all inputs that exist, for every model at the same time. Flask API will have this working normally. Just make sure you have the model loaded and everything should be good. Extensions are planned for the near future, I just have to make sure extensions will be forwards compatible, so i need to make sure i'm using good names for everything, as you can't really change this in the future without breaking at least a few plugins.
